### PR TITLE
feat: add osx and plugins artifacts

### PR DESCRIPTION
**Pull request by clauBv23**
## Description

Create a new package for publishing the oss and the plugins abis. 

Task ID: [OS-1445](https://aragonassociation.atlassian.net/browse/OS-1445)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I created tasks to update dependent repositories (OSx, Plugins)
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

[OS-1445]: https://aragonassociation.atlassian.net/browse/OS-1445?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: artifacts/generateAbis.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+source .env
+export ALCHEMY_API_KEY=$ALCHEMY_API_KEY
+
+
```
Comment (novaknole) [inline artifacts/generateAbis.sh:6]: Probably better to exit and show an error if alchemy key is not provided.

File: artifacts/generateAbis.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        exit 1
+    fi 
+
+    yarn wagmi generate
+    
+    mv generated/abis.ts $current_dir/src/abis/${REPO_NAME}-abis.ts
+
+    cd $current_dir/$CLONE_DIR
+
+done <<< "$REPOS"
```
Comment (novaknole) [inline artifacts/generateAbis.sh:63]: EOF. Put an enter to add one empty line.

File: artifacts/src/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from "./abis";
```
Comment (novaknole) [inline artifacts/src/index.ts:1]: EOF

File: artifacts/wagmi.foundry.config.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export default defineConfig({
+  // using abi instead of generated to avoid being ignored by git
+  out: 'generated/abis.ts',
+  plugins: [
+    foundry({
+      project: './',
+      exclude: ['**/test/**', '**/mocks/**'],
+    }),
+  ],
+});
```
Comment (novaknole) [inline artifacts/wagmi.foundry.config.ts:13]: EOF

File: artifacts/wagmi.hardhat.config.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  // using abi instead of generated to avoid being ignored by git
+  out: 'generated/abis.ts',
+  plugins: [
+    hardhat({
+      project: './',
+      artifacts: 'REPLACE_ARTIFACTS_SOURCE',
+      exclude: ['**/test/**', '**/mocks/**'],
+    }),
+  ],
+});
```
Comment (novaknole) [inline artifacts/wagmi.hardhat.config.ts:14]: EOF

**Review by novaknole — COMMENTED** [review]
A couple of easy changes required.

File: artifacts/generateAbis.sh (hunk 2 above)
Comment (clauBv23) [reply artifacts/generateAbis.sh:63]: That's because prettier is not working let me fix that. 

good catch

File: artifacts/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+## v0.1.0-alpha.1
+
+## Added
+
+- Add osx's contracts artifacts.
+- Add admin plugin's contracts artifacts.
+- Add multisig plugin's contracts artifacts.
+- Add token voting plugin's contracts artifacts.
+- Add staged proposal processor plugin's contracts artifacts.
```
Comment (Rekard0) [inline artifacts/CHANGELOG.md:16]: perhaps we need to write the corresponding versions

**Review by Rekard0 — APPROVED** [review]
LGTM, please don't forget to follow up with: https://aragonassociation.atlassian.net/browse/OS-1457

**Review by novaknole — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**